pkgs/build-support/bintools-wrapper/setup-hook.sh

```diff
@@ -61,9 +62,8 @@ do
     if
         PATH=$_PATH type -p "@targetPrefix@${cmd}" > /dev/null
     then
-        upper_case="$(echo "$cmd" | tr "[:lower:]" "[:upper:]")"
-        export "${role_pre}${upper_case}=@targetPrefix@${cmd}";
-        export "${upper_case}${role_post}=@targetPrefix@${cmd}";
+        export "${role_pre}${cmd^^}=@targetPrefix@${cmd}";
+        export "${cmd^^}${role_post}=@targetPrefix@${cmd}";
```

Comment:
I originally did something like this, but people complained on darwin that nix-shell uses the ambient bash which is old as hell. But Apple is switching to zsh anyways I think? And also I think supporting ambient bashes is stupid; fix the usability a different way.

Reply:
I don't think we need to worry about builtin bash anyway. It can be an issue with bootstrapping, but we just need to tell people to build a newer Bash before they build Nix from source.

Reply:
Oh! running `nix-shell` shows that it is not impure anymore yay!!!

Reply:
I think this was only ever a problem in the impure stdenv though. It would be good to document our minimum Bash version though. I think all of these new features were added in Bash 4?

Reply:
#71625